src/core/send.c

```diff
@@ -1018,6 +1018,46 @@ QuicSendGetNextStream(
     return NULL;
 }
 
+BOOLEAN
+CxPlatIsRouteReady(
+    _In_ QUIC_CONNECTION *Connection
+) {
+    QUIC_PATH* Path = &Connection->Paths[0];
+    //
+    // Make sure the route is resolved before sending packets.
+    //
+    if (Path->Route.State == RouteResolved) {
+        return TRUE;
+    }
+
+    //
+    // We need to set the path challenge flag back on so that when route is resolved,
+    // we know we need to continue to send the challenge.
+    //
+    CXPLAT_DBG_ASSERT(Path->IsActive);
+    if (Path->Route.State == RouteUnresolved) {
```

Comment:
```suggestion
    if (Path->Route.State == RouteUnresolved || Path->Route.State == RouteSuspected) {
```
We need to copy the logic originally in the normal send path, which includes the `RouteSuspected` case.